src/Core/System/Currency/Api/ChangeCurrencyController.php

```diff
@@ -0,0 +1,143 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Currency\Api;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Adapter\Cache\CacheClearer;
+use Shopware\Core\Framework\DataAbstractionLayer\DefinitionInstanceRegistry;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\PriceField;
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\System\Currency\Message\RecalculatePricesForCurrencyMessage;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"api"})
+ */
+class ChangeCurrencyController
```

Comment:
why so specific? what if we need more actions for the currency?

CurrencyController would be sufficient, I think